astropy/io/fits/_tiled_compression/codecs.py

```diff
@@ -34,9 +34,64 @@ class Codec:
     "Rice1",
     "PLIO1",
     "HCompress1",
+    "NoCompress",
 ]
 
 
+def _as_big_endian_array(data):
+    return data.astype(np.asarray(data).dtype.newbyteorder(">"))
+
+
+def _as_native_endian_array(data):
+    if data.dtype.isnative:
+        return data
+    else:
+        return data.astype(np.asarray(data).dtype.newbyteorder("="))
```

Comment:
Maybe those functions could be used more widely in the `io.fits` code, where we probably have various ways to do the conversion, so maybe they could go in `fits.util` (but that can also be done later).

Reply:
I agree, let's do it later since we should probably at the same time scan through the io.fits code to see where we can use these.